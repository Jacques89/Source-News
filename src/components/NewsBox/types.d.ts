type NewsBoxProps = {
  article: {
    title: string
    img: string
    content: string
    source_url: string
  }
}