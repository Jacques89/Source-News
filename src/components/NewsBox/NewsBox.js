import React, { Component } from 'react'
import '../../styles/NewsBox.scss'

export default class NewsBox extends Component {
  render () {
    console.log(this.props)
    let article = this.props.article
    let title = article.title
    let description = article.description
    let image = article.image
    let url = article.url

    return (
      <article>
        <h3 className="title">{title}</h3>
        <div className="image" style={{ backgroundImage: `url('${image}')` }}></div>
        <p className="description">
          {description}
        </p>
        <p className="link">
          <a href={url}>Continue reading</a>
        </p>
      </article>
    )
  }
}