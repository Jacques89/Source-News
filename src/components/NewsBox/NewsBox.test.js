import React from 'react'
import NewsBox from './NewsBox'

import { render, screen, act, fireEvent } from '@testing-library/react' 

describe('NewsBox', () => {
  const props = {
    article: 'article-title',
    image: 'news.png',
    description: 'News description',
    source: 'https://externalsource.com'
  }
  
  const wrapper = <NewsBox {...props} />

  it('renders the NewsBox component correctly', () => {
    const NewsBoxComponent = render(wrapper)
    expect(NewsBoxComponent).toMatchSnapshot()
  })
  
  it('renders with an article-title', async() => {
    await act(async() => render(wrapper))
    expect(screen.getByTestId('article-title')).toBeTruthy()
  })

  it('renders with a background image', async() => {
    await act(async() => render(wrapper))
    expect(screen.getByTestId('article-image')).toBeTruthy()
  })

  it('renders with article content', async() => {
    await act(async() => render(wrapper))
    expect(screen.getByTestId('article-content')).toBeTruthy()
  })

  it('renders with a link to an external site', async() => {
    await act(async() => render(wrapper))
    expect(screen.getByTestId('external-link')).toBeTruthy()
  })

  it('should navigate to an external site when link is clicked', () => {
    const { getByText } = render(<a href="https://sourcetest.com/">Continue reading</a>)

    const link = getByText('Continue reading')

    fireEvent.click(link)

    expect(getByText('Continue reading').href).toBe('https://sourcetest.com/')
  })
})