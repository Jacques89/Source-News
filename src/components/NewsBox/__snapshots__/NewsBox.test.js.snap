// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsBox renders the NewsBox component 1`] = `
<article
  data-testid="article"
>
  <h3
    class="title"
    data-testid="article-title"
  />
  <div
    class="image"
    data-testid="article-image"
    style="background-image: url(undefined);"
  />
  <p
    class="description"
    data-testid="article-content"
  />
  <p
    class="link"
  >
    <a
      data-testid="external-link"
    >
      Continue reading
    </a>
  </p>
</article>
`;
